File: TCP  IP Server Client Example Part 7 (Chat Application)/[C#][VB]-TCP  IP Server Client Example Part 7 (Chat Application)/C# and VB/ChatClient/FrmRegister.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ChatClient
{
    public partial class FrmRegister : Form
    {
        public string RegisteredUserEmail { get; set; }
        public string RegisteredUserPassword { get; set; } // We have hard coded all passwords to pass to speed up debugging

        public FrmRegister()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Allow user to Register and Login Simultaneously
        /// Add your own Error Validation
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void btnCancel_Click(object sender, EventArgs e)
        {
            this.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.Close();
        }

        /// <summary>
        /// Allow user to Register and Login Simultaneously
        /// Add your own Error Validation
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void btnLogin_Click(object sender, EventArgs e)
        {
            if (this.tbEmailAddress.Text.Length > 0 && this.tbPassword.Text.Length > 0 && this.tbPassword.Text == this.tbConfirmPassword.Text)
            {
                this.RegisteredUserEmail = this.tbEmailAddress.Text;
                this.RegisteredUserPassword = this.tbPassword.Text.Replace(",", ""); // Remove ',' (commas) from passwords as it could invalidate with our messages
                this.DialogResult = System.Windows.Forms.DialogResult.OK;
                this.Close();
            }
            else
            {
                MessageBox.Show("Registration Details are not Valid");
            }
        }
    }
}

File: TCP  IP Server Client Example Part 7 (Chat Application)/[VB]-TCP  IP Server Client Example Part 7 (Chat Application)/VB/ChatClient/FrmRegister.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace ChatClient
	Partial Public Class FrmRegister
		Inherits Form

		Public Property RegisteredUserEmail() As String
		Public Property RegisteredUserPassword() As String

		Public Sub New()
			InitializeComponent()
		End Sub

		''' <summary>
		''' Allow user to Register and Login Simultaneously
		''' Add your own Error Validation
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub btnCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCancel.Click
			Me.DialogResult = System.Windows.Forms.DialogResult.Cancel
			Me.Close()
		End Sub

		''' <summary>
		''' Allow user to Register and Login Simultaneously
		''' Add your own Error Validation
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub btnLogin_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnLogin.Click
			If Me.tbEmailAddress.Text.Length > 0 AndAlso Me.tbPassword.Text.Length > 0 AndAlso Me.tbPassword.Text = Me.tbConfirmPassword.Text Then
				Me.RegisteredUserEmail = Me.tbEmailAddress.Text
				Me.RegisteredUserPassword = Me.tbPassword.Text.Replace(",", "") ' Remove ',' (commas) from passwords as it could invalidate with our messages
				Me.DialogResult = System.Windows.Forms.DialogResult.OK
				Me.Close()
			Else
				MessageBox.Show("Registration Details are not Valid")
			End If
		End Sub
	End Class
End Namespace
